### PR TITLE
Scripts for processing the pynublado outputs into deepnublado inputs

### PR DESCRIPTION
**Pull request by raulorteg**
Addtions :sparkles: 
-------------------------
* processing script to concatenate pynublado's binaries into .csv files to be fed to the deepnublado pipeline ```scripts/process_pynublado_binaries.py```
* splitting script to create the train/test partitioning from the .csv files ```scripts/train_test_split.py```
* settings file for common/global configuration/settings use across different scripts. ```src/settings.py```
* Pinned requirements to exact versions, so we can dockerize it easily later on. ```requirements.txt```

Documentation :book:
--------------------------
The usage of the scripts added is documented in ```docs/processing_pynublado.md```, I added the documentation of their usage as a separate file in docs in preparation for when we build the sphinx docs as we did with pynublado (https://raulorteg.github.io/pyNublado/)

Code formatting and style :art: 
------------------------------------
I used black==22.10.0  and isort==5.10.1 for code formatting and sorting the import, I will create an issue so we can discuss a contributing guidelines.

Example outputs:
-----------------------
Ouput from ```scripts/process_pynublado_binaries.py``` :
image

Output from ```scripts/train_test_split.py```:
image

### CODE REVIEW

**Comment by eor** [comment]
Hi Raul, looks like you did quite a bit of work here already. I have a few questions / comments:

1. Do we need a classifier network to predict the simulation status for a given input parameter vector? Doesn't the dataset that the pyNublado parser generates only contain successful runs?
2. Conceptually, why are we converting the binary data to csv files when they will be re-converted to a binary format for the actual ML part? Are there any other processing steps that need to happen or do you want to have greater flexibility regarding, e.g. the filtering of certain lines. I thought the pandas data frames would be up to the job.
3.  Regarding the picture above: Why is the memory footprint of the Regressor dset only 30 MB when the original dataset contained gigabytes of binary data?
4. I would pin version number of packages after we are done and ready to publish to still receive (security) updates during the development phase
5. Shouldn't the split of the data set also contain a validation set? I think I only saw `_train` and `_test` sets.
6. Regarding the config file: I think it's good to have a central place for settings but think that hyper-parameters should not be static and rather be set via command line arguments to allow for easy and scriptable exploration of the hyper-parameter space. Maybe the values in the file could be defaults?
7. The term `Regressor` here refers only to the emission line data as far as I could see. What about the other data that will (hopefully) also be inferred (continuum / cooling)? Will they be part of the same csv file or read separately?

Maybe we can have a chat about some of  these points?

**Comment by raulorteg** [comment]
1. We parse all runs, even if they have problems, but we take note if they have problems and which kind of problems (convergence, negative populations, ...).  This allows us to then train a classifier to pre-filter user inputs, as a way of raising warnings if the given inputs would be likely to fail under CLOUDY.
2. & 3. These resulting .csv files can be intermediate data products. Here we remove a lot of data that is not needed for the surrogate models (strictly inputs & outputs) thus reducing the amount of disk memory needed to store them. Given their current size they also allow for manual inspection, which might be helpful at this stage of the project.
4. More of a personal choice for me, I prefer to pin them now. But I can do >= instead of == in the requirements.txt and then we pin them at the end if you prefer.
5. Added now in [ceaf13e](https://github.com/raulorteg/deepnublado/pull/2/commits/ceaf13ed801641c9877e6a0beab9e48fc3885edf)
6. I prefer not to overpopulate the argparser with too many options that make it unusable, I think some options to be given from the command line can be useful but if the number of hyperparameters or options becomes too large I think the best option is to have them in a .py, .json or .yaml file and load them from there. We can change that later on.
7. Maybe it could be called regressor_emission but since right now we do not parse the cooling tables I dont think we need to change the name, we can do that later on if we parse the cooling tables.

**Review by eor — APPROVED** [review]
Look fine to me. Thanks Raul!
